mlflow/store/tracking/file_store.py

```diff
@@ -1319,3 +1326,372 @@ def _read_helper(root, file_name, attempts_remaining=2):
+            timestamp_ms: Start time of the trace, in milliseconds since the UNIX epoch.
+            request_metadata: Metadata of the trace.
+            tags: Tags of the trace.
+
+        Returns:
+            The created TraceInfo object.
+        """
+        request_id = request_metadata.get(SpanAttributeKey.REQUEST_ID)
+        if request_id is None:
+            raise MlflowException("Request ID is required to start a trace.", INTERNAL_ERROR)
+        _validate_experiment_id(experiment_id)
+        experiment_dir = self._get_experiment_path(
+            experiment_id, view_type=ViewType.ACTIVE_ONLY, assert_exists=True
+        )
+        mkdir(experiment_dir, FileStore.TRACES_FOLDER_NAME)
+        traces_dir = os.path.join(experiment_dir, FileStore.TRACES_FOLDER_NAME)
+        mkdir(traces_dir, request_id)
+        trace_dir = os.path.join(traces_dir, request_id)
+        trace_info = TraceInfo(
+            request_id=request_id,
+            experiment_id=experiment_id,
+            timestamp_ms=timestamp_ms,
+            execution_time_ms=None,
+            status=TraceStatus.IN_PROGRESS,
+            request_metadata=request_metadata,
+            tags=tags,
+        )
+        self._save_trace_info(trace_info, trace_dir)
+        return trace_info
+
+    def _save_trace_info(self, trace_info: TraceInfo, trace_dir, overwrite=False):
+        """
+        TraceInfo is saved into `traces` folder under the experiment, each trace
+        is saved in the folder named by its request_id.
+        `request_metadata` and `tags` folder store their key-value pairs such that each
+        key is the file name, and value is written as the string value.
+        Detailed directories structure is as below:
+        | - experiment_id
+        |   - traces
+        |     - request_id1
+        |       - meta.yaml
+        |       - request_metadata
+        |       - tags
+        |     - request_id2
+        |     - ...
+        |   - run_ids ...
+        """
+        # Save basic trace info to META_DATA_FILE_NAME
+        trace_info_dict = self._convert_trace_info_to_dict(trace_info)
+        write_yaml(trace_dir, FileStore.META_DATA_FILE_NAME, trace_info_dict, overwrite=overwrite)
+        # Save request_metadata to its own folder
+        self._write_dict_to_trace_sub_folder(
+            trace_dir,
+            FileStore.TRACE_REQUEST_METADATA_FOLDER_NAME,
+            trace_info.request_metadata,
+        )
+        # Save tags to its own folder
+        self._write_dict_to_trace_sub_folder(
+            trace_dir, FileStore.TRACE_TAGS_FOLDER_NAME, trace_info.tags
+        )
+
+    def _convert_trace_info_to_dict(self, trace_info: TraceInfo):
+        """
+        Convert trace info to a dictionary for persistence.
+        Drop request_metadata and tags as they're saved into separate files.
+        """
+        trace_info_dict = asdict(trace_info)
+        trace_info_dict["status"] = trace_info.status.value
```

Comment:
qq: What happens if we remove this conversion? If it cause some issue, can we have this handling within `TraceInfo` entity (so we don't have to replicate same for other places)?

Reply:
It can't be written into yaml file, as the value itself is an object. I'll add a helper function in TraceInfo instead.